src/app/account.service.ts

```diff
@@ -74,6 +83,47 @@ export class AccountService {
     }
   }
 
+  getDerivedUser(publicKey: string, blockHeight: number): DerivedUserInfo{
+    const privateUser = this.getPrivateUsers()[publicKey];
+
+
+    this.entropyService.setNewTemporaryEntropy();
+    const derivedMnemonic = this.entropyService.temporaryEntropy?.mnemonic;
+    const derivedKeychain = this.cryptoService.mnemonicToKeychain(derivedMnemonic);
+    const derivedSeedHex = this.cryptoService.keychainToSeedHex(derivedKeychain);
+    const derivedPrivateKey = this.cryptoService.seedHexToPrivateKey(derivedSeedHex);
+    const derivedPublicKey = this.cryptoService.privateKeyToDeSoPublicKey(derivedPrivateKey, privateUser.network);
+
+    // By default we authorize this derived key for 10,000 blocks.
+    const expirationBlock = blockHeight + 10000;
+
+    const expirationBlockBuffer = uint64ToBufBigEndian(expirationBlock);
+    const derivedPublicKeyBuffer = derivedPrivateKey.getPublic().encode('array', true);
+    const accessHash = sha256.x2([...derivedPublicKeyBuffer, ...expirationBlockBuffer]);
+    const accessSignature = this.signingService.signBurn(privateUser.seedHex, [accessHash])[0];
```

Comment:
If this method is being used to sign things other than burns then we should rename it to something like `signHashes`